### PR TITLE
[fix] Fix alltoall strategy dispatch TypeError when quant_mode is None

### PR DESCRIPTION
**Pull request by 1329009851**
## Motivation

fixed issue: https://github.com/sgl-project/sgl-kernel-npu/issues/647
PR #600 added quant_mode to buffer.dispatch() forwarding it to the normal strategy, but AlltoAllNormalCommStrategy.dispatch() lacked the parameter, causing TypeError whenever DEEP_USE_MODE=alltoall was set.

## Modifications

- Add quant_mode param to AlltoAllNormalCommStrategy.dispatch()
- Validate quant_mode against the shared VALID_QUANT_MODES set
- Map quant_mode=='int8' to npu_dynamic_quant, else bf16
- Raise NotImplementedError for unsupported FP8/FP4 modes
- Fall back to DEEP_NORMAL_MODE_USE_INT8_QUANT env var when quant_mode is None
- Sync quant_mode into NormalEPCommStrategy abstract dispatch signature

## Testing

The alltoall test case is not fully developed, and the complete test case has not been successfully executed.
The test result shows that the code is successfully executed in the dispatch, indicating that the modification has taken effect.
image

## Checklist

- [x] Format your code.
- [x] Add unit tests.
- [x] Update documentation.
- [x] Provide accuracy and speed benchmark results.

### CODE REVIEW

File: python/deep_ep/deep_ep/strategies/normal_strategy.py [hunk 1]
```diff
@@ -558,6 +558,7 @@ def dispatch(
         async_finish: bool = False,
         allocate_on_comm_stream: bool = False,
         dispatch_wait_recv_cost_stats: Optional[torch.Tensor] = None,
+        quant_mode: Optional[str] = None,
```
Comment (kaniel-outis) [inline python/deep_ep/deep_ep/strategies/normal_strategy.py:557]: 合入前补充Testing

File: python/deep_ep/deep_ep/strategies/normal_strategy.py [hunk 2]
```diff
@@ -577,6 +578,32 @@ def dispatch(
         ]
         global_tokens_indices = layout["global_tokens_indices"]
 
+        VALID_QUANT_MODES = {
```
Comment (zuje123) [inline python/deep_ep/deep_ep/strategies/normal_strategy.py]: 多个地方使用，建议单独提取出来

**Review by zuje123 — DISMISSED** [review]

**Review by zuje123 — APPROVED** [review]

File: python/deep_ep/deep_ep/strategies/normal_strategy.py [hunk 3]
```diff
@@ -11,7 +11,11 @@
 import torch_npu
 from deep_ep_cpp import EventHandle
 
-from ..ep_strategy import NormalEPCommStrategy, register_normal_strategy
+from ..ep_strategy import (
```
Comment (kaniel-outis) [inline python/deep_ep/deep_ep/strategies/normal_strategy.py:14]: 链接对应的issue

**Review by kaniel-outis — APPROVED** [review]
